bindings/azure/cosmosdb/cosmosdb.go

```diff
@@ -73,36 +75,56 @@ func (c *CosmosDB) Init(metadata bindings.Metadata) error {
 	// this allows us to provide the most flexibility in the request document sent to this binding
 	config.IdentificationHydrator = nil
 	config.WithAppIdentifier("dapr-" + logger.DaprVersion)
-	client := documentdb.New(m.URL, config)
 
-	dbs, err := client.QueryDatabases(&documentdb.Query{
-		Query: "SELECT * FROM ROOT r WHERE r.id=@id",
-		Parameters: []documentdb.Parameter{
-			{Name: "@id", Value: m.Database},
-		},
-	})
-	if err != nil {
-		return err
-	} else if len(dbs) == 0 {
-		return fmt.Errorf("database %s for CosmosDB state store not found", m.Database)
-	}
+	// Retries initializing the client if a TooManyRequests error is encountered
+	bo := backoff.NewExponentialBackOff()
+	bo.InitialInterval = 2 * time.Second
+	bo.MaxElapsedTime = 5 * time.Minute
+	err = backoff.RetryNotify(func() (err error) {
```

Comment:
Consider using the `github.com/dapr/kit/retry` wrapper for `NotifyRecover` so that it also logs success after retry, it'll help with logging consistency across components.

Similarly for the state component.

Reply:
This was discussed with @pkedy  beforehand. I specifically inquired about kit/retry. We actually want to log the warnings here and hence didn't use the retry package.